src/liblibc/lib.rs

```diff
@@ -146,10 +147,15 @@ pub use funcs::bsd43::*;
 #[link(name = "m")]
 extern {}
 
-#[cfg(all(target_env = "musl", not(test)))]
+#[cfg(all(target_env = "musl", not(feature = "cargo-build"), not(test)))]
 #[link(name = "c", kind = "static")]
 extern {}
 
+// When performing a cargo build with musl, we might not be able to find libc.a
+// Fortunately, it's already been statically included in the liblibc archive.
+#[cfg(all(target_env = "musl", feature = "cargo-build", not(test)))]
+extern crate libc;
```

Comment:
This unfortunately definitely needs to be left out because the liblibc library on crates.io needs to be able compile on stable Rust. I don't think this is necessary, however, because liblibc is included in the standard library which is transitively linked to this library, so this directive shouldn't be necessary I believe.